### PR TITLE
Documented All Pallets

### LINKED ISSUE #80: Document Pallet Tickets
This pallet has mostly internal stuff but good to have some internal docs anyway

### PR DESCRIPTION
**Pull request by aang114**
1. Added documentation for all the pallets
2. Created a new workflow job which updates this repository's GitHub Pages whenever the `devel` branch is updated

### CODE REVIEW

File: pallets/detach/src/lib.rs [hunk 1]
```diff
@@ -524,16 +589,25 @@ pub mod pallet {
...
-												public: account.public.clone(),
+
+
+									// `send_unsigned_transaction` is returning a type of `Option<(Account<T>, Result<(), ()>)>`.
+									//   The returned result means:
+									//   - `None`: no account is available for sending transaction
+									//   - `Some((account, Ok(())))`: transaction is successfully sent
+									//   - `Some((account, Err(())))`: error occurred when sending the transaction
+									if let Err(e) = Signer::<T, T::AuthorityId>::any_account() // `Signer::<T, T::AuthorityId>::any_account()` is the signer that signs the payload
+										.send_unsigned_transaction( // INC questo block
```
Comment (sinkingsugar) [inline pallets/detach/src/lib.rs:600]: maybe remove?

File: pallets/detach/src/lib.rs [hunk 2]
```diff
@@ -524,16 +589,25 @@ pub mod pallet {
...
+											// this line is to prepare and return payload to be used
+											|account| DetachInternalData { // `account` is the account `Signer::<T, T::AuthorityId>::any_account()`
+												public: account.public.clone(), // 问Gio what is account.public and why is it supposed to be in FragKey
```
Comment (sinkingsugar) [inline pallets/detach/src/lib.rs]: remove or clarify

File: pallets/detach/src/lib.rs [hunk 3]
```diff
@@ -288,8 +317,10 @@ pub mod pallet {
 		}
 	}
 
+	/// Define some logic that should be executed regularly in some context, for e.g. `on_initialize`.
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		/// During the block finalization phase, the **list** in **DetachRequests** is **stored in the Offchain DB** under the **index "fragments-detach-requests"**. Afterwhich, **DetachRequests** is **cleared**
```
Comment (sinkingsugar) [inline pallets/detach/src/lib.rs]: Afterwhich -> After which I think

File: pallets/protos/src/mock.rs [hunk 4]
```diff
@@ -147,3 +167,46 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
 	let t = frame_system::GenesisConfig::default().build_storage::<Test>().unwrap();
 	t.into()
 }
+
```
Comment (sinkingsugar) [inline pallets/protos/src/mock.rs]: what's this?

**Review by sinkingsugar — APPROVED** [review]

File: pallets/protos/src/mock.rs (hunk 4 above)
Comment (aang114) [reply pallets/protos/src/mock.rs]: removed

File: pallets/detach/src/lib.rs (hunk 3 above)
Comment (aang114) [reply pallets/detach/src/lib.rs]: changed

File: pallets/detach/src/lib.rs (hunk 2 above)
Comment (aang114) [reply pallets/detach/src/lib.rs]: removed

**Pull request merged**